src/lib.rs

```diff
@@ -70,12 +71,42 @@ fn shrink<T>(slice: &mut &mut [T], len: usize) {
 fn is_token(b: u8) -> bool {
     b > 0x1F && b < 0x7F
 }
+
 macro_rules! byte_map {
     ($($flag:expr,)*) => ([
         $($flag != 0,)*
     ])
 }
 
+// ASCII codes to accept URI string.
+// i.e. A-Z a-z 0-9 !#$%&'*+-._();:@=,/?[]~
+// TODO: Make a stricter checking for URI string?
+static URI_MAP: [bool; 256] = byte_map![
+    0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+	0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
```

Comment:
Is this indentation difference intentional?

Reply:
My fault, I should check my editor config.